app/experimenter/nimbus-ui/src/components/PageResults/GraphsWeekly/index.tsx

```diff
@@ -0,0 +1,129 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import React, { useState } from "react";
+import Collapse from "react-bootstrap/Collapse";
+import vegaEmbed, { VisualizationSpec } from "vega-embed";
+import {
+  BRANCH_COMPARISON,
+  BRANCH_COMPARISON_TITLE,
+} from "../../../lib/visualization/constants";
+import { lineGraphConfig } from "../../../lib/visualization/graphConfig";
+import {
+  AnalysisDataWeekly,
+  BranchDescription,
+  FormattedAnalysisPoint,
+} from "../../../lib/visualization/types";
+import { ReactComponent as CollapseMinus } from "./minus.svg";
+import { ReactComponent as ExpandPlus } from "./plus.svg";
+
+/**
+ * We should only graph if there are at least 2 weeks of data
+ * because otherwise the graph is just a single point and useless.
+ *
+ * So we return empty mergedBranchData to avoid graphing when there
+ * isn't enough data
+ */
+const getMergedBranchData = (
```

Comment:
Is this the kind of thing that can be prepared on the backend?

Reply:
Yea, some of this could be factored out into the backend. I'll incorporate that in the backend refactor. In a couple of sprints likely